bn_deprecated.c

```diff
@@ -230,18 +230,85 @@ mp_err mp_n_root(const mp_int *a, mp_digit b, mp_int *c)
 }
 #endif
 #ifdef BN_MP_TORADIX_N_C
+/* Just C&P for simplicity */
 mp_err mp_toradix_n(const mp_int *a, char *str, int radix, int maxlen)
 {
-   if (maxlen < 0) {
+   mp_err res;
+   size_t digs;
+   mp_int  t;
+   mp_digit d;
+   char   *_s = str;
+
+   /* check range of the maxlen, radix */
+   if ((maxlen < 2) || (radix < 2) || (radix > 64)) {
       return MP_VAL;
    }
-   return mp_to_radix(a, str, (size_t)maxlen, radix);
+
+   /* quick out if its zero */
+   if (mp_iszero(a) == MP_YES) {
+      *str++ = '0';
+      *str = '\0';
+      return MP_OKAY;
+   }
+
+   if ((res = mp_init_copy(&t, a)) != MP_OKAY) {
+      return res;
+   }
+
+   /* if it is negative output a - */
+   if (t.sign == MP_NEG) {
+      /* we have to reverse our digits later... but not the - sign!! */
+      ++_s;
+
+      /* store the flag and mark the number as positive */
+      *str++ = '-';
+      t.sign = MP_ZPOS;
+
+      /* subtract a char */
+      --maxlen;
+   }
+   digs = 0;
+   while (mp_iszero(&t) == MP_NO) {
+      if (--maxlen < 1) {
+         /* no more room */
+         break;
+      }
+      if ((res = mp_div_d(&t, (mp_digit) radix, &t, &d)) != MP_OKAY) {
+         mp_clear(&t);
+         return res;
+      }
+      *str++ = mp_s_rmap[d];
+      ++digs;
+   }
+
+   /* reverse the digits of the string.  In this case _s points
+    * to the first digit [exluding the sign] of the number
+    */
+   s_mp_reverse((unsigned char *)_s, digs);
+
+   /* append a NULL so the string is properly terminated */
+   *str = '\0';
+
+   mp_clear(&t);
+   return MP_OKAY;
+
 }
 #endif
 #ifdef BN_MP_TORADIX_C
+/* TODO: change to "size_t" when the type of "mpint.size" changes to "size_t" */
+static int s_mp_strlen(const char *s)
+{
+   const char *p;
+   p = s;
+   while (*p != '\0') {
+      p++;
+   }
+   return (int)(p - s);
+}
 mp_err mp_toradix(const mp_int *a, char *str, int radix)
 {
-   return mp_to_radix(a, str, SIZE_MAX, radix);
+   MP_FREE_BUFFER(str, (size_t)(s_mp_strlen(str) + 1));
```

Comment:
> What if it's located on the stack?

It will crash and burn and die with a whimper as it is fitting for a deprecated function! ;-)

Actually: `helper.pl` complained so I made it a Q&vD at that time (which didn't hinder `helper.pl` from complaining again ;-) ); should have put a `TODO` comment there at least, sorry.

Make an exemption for `bn_deprectated.c` or use an internal `memcpy`?
Something cheap and small like e.g.:

```c
static void *s_memcpy (void *dst, const void *src, size_t len)
{
   char *d = (char *) dst;
   char *s = (char *) src;

   void *start = dst;

   while (len--)  {
       *d = *s;
       d++; s++;
   }
   return start;
}
```

or similar would suffice, so it isn't much work and doesn't use much memory.
There are also the `*_bin*` functions which will have similar (and other) problems in `bn_deprectated.c`, so it won't be a single-use function.
Yeah, I think I'll go with an internal function if nobody complains.